actionmailer/lib/action_mailer/railtie.rb

```diff
@@ -41,8 +41,9 @@ class Railtie < Rails::Railtie # :nodoc:
         register_preview_interceptors(options.delete(:preview_interceptors))
         register_observers(options.delete(:observers))
 
-        if delivery_job = options.delete(:delivery_job)
-          self.delivery_job = delivery_job.constantize
+        delivery_job = options.delete(:delivery_job)
+        ActiveSupport.on_load(:active_job) do
+          ::ActionMailer::Base.delivery_job ||= delivery_job.present? ? delivery_job.constantize : ::ActionMailer::MailDeliveryJob
         end
```

Comment:
Similar to my other comment, are there potential scenarios where `ActionMailer::Base.delivery_job` will not be set because nothing else (besides `ActionMailer::MailDeliveryJob` itself) has referenced / loaded `ActiveJob::Base`?